packages/common-common/src/rabbitmq/util.ts

```diff
@@ -0,0 +1,75 @@
+import fetch from "node-fetch";
+import {RascalQueues, RascalExchanges, RascalRoutingKeys} from './types';
+
+
+/*
+For more information on any of these routes and their behaviour/return values check:
+https://rawcdn.githack.com/rabbitmq/rabbitmq-server/v3.11.1/deps/rabbitmq_management/priv/www/api/index.html
+ */
+
+/**
+ * This function publishes a message containing the given data to the given exchange. In the event of a 400 Bad Request
+ * error ensure ALL nested objects in the data object are stringified BEFORE calling this function.
+ * @param rabbitMQUri The URI of the RabbitMQ Management API instance to connect to
+ * @param exchangeName The name of the exchange to publish the message to
+ * @param routingKey The routing key which defines what queue a message is routed to from the given exchange
+ * @param data The object to publish (WARNING: all nested objects must be individually json stringified)
+ * @returns {routed: boolean} If routed is false the message was not queued and if routed is true the message was queued
+ */
+export async function publishRmqMsg(
+  rabbitMQUri: string,
+  exchangeName: RascalExchanges,
+  routingKey: RascalRoutingKeys,
+  data: any
+): Promise<{routed: boolean}> {
+  const publishBody = JSON.stringify({
+    properties: {content_type: "application/json"},
+    routing_key: routingKey,
+    payload: JSON.stringify(data),
+    props: {
+      "content_type": "application/json"
+    },
+    payload_encoding: "string"
+  });
+  const publishResult = await fetch(`${rabbitMQUri}/exchanges/%2f/${exchangeName}/publish`, {
+    method: 'post',
+    body: publishBody,
+    headers: {'Content-Type': 'application/json'}
+  });
+  return await publishResult.json();
+}
+
+/**
+ * This function fetches queue stats for the given queue.
+ * @param rabbitMQUri The URI of the RabbitMQ Management API instance to connect to
+ * @param queueName
+ */
+export async function getQueueStats(rabbitMQUri: string, queueName: RascalQueues): Promise<any> {
+  let result = await fetch(`${rabbitMQUri}/queues/%2f/${queueName}`);
+  return await result.json();
+}
+
+/**
+ * This functions retrieves a number of messages from the specified RabbitMQ queue
+ * @param rabbitMQUri The URI of the RabbitMQ Management API instance to connect to
+ * @param queueName The name of the queue to retrieve messages from
+ * @param requeue [Default true] A boolean variable indicating whether the message(s) should be left in the queue
+ * @param numMessages The number of messages to fetch from the queue
+ */
```

Comment:
Same with all these comments, it makes reading the code more cumbersome. Useful comments, but I don't think we should make a habit to write these param types in. 

The types themselves provide this kind of documentation 👍

Reply:
The types would tell you that `numMessages` is a number but not what the number indicates or is intended for. I would actually argue this format of documentation should be on nearly all functions we have in the repo so that we don't have to figure out what each argument is supposed to do when calling the function.